xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -41,33 +40,35 @@ CollTfCollLenFeature::get_values() const
 {
     LOGCALL(API, std::vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
 
-    Query query = Feature::internal->feature_query;
-    map<string, long int> coll_tf = Feature::internal->collection_termfreq();
-    map<string, long int> coll_len = Feature::internal->collection_length();
-
     vector<double> values;
     double value = 0;
 
-    for (Xapian::TermIterator qt = query.get_terms_begin(); qt != query.get_terms_end(); ++qt) {
+    for (Xapian::TermIterator qt = feature_query.get_unique_terms_begin();
+	 qt != feature_query.get_terms_end(); ++qt) {
 	if ((*qt).substr(0, 1) == "S" || (*qt).substr(1, 1) == "S")
-	    value += log10(1 + ((double)coll_len["title"] / (double)(1 + coll_tf[*qt])));
+	    value += log10(1 + ((double)collection_length.find("title")->second
```

Comment:
`collection_length.find("title")->second` is going to be the same for every iteration - stick in it a variable before the loop and that'll make this overly long expression shorter and easier to read, and may also be faster (I suspect the compiler won't be able to see that this lookup can be hoisted).

A similar comment applies to the loops below too.